bundles/binding/org.openhab.binding.gc100ir/src/main/java/org/openhab/binding/gc100ir/lib/TCPIPSocket.java

```diff
@@ -0,0 +1,195 @@
+/**
+ * Copyright (c) 2010-2013, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.gc100ir.lib;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Manages the connection with the TCP/IP socket for the specified device
+ * attached to the GC-100.
+ * 
+ * @author Parikshit Thakur & Team
+ * @since 1.6.0
+ */
+
+class TCPIPSocket {
```

Comment:
please rename this class since it does not provide the implementation of a Socket itself